src/bin/pg_autoctl/pgsetup.c

```diff
@@ -477,16 +479,19 @@ get_pgpid(PostgresSetup *pgSetup, bool pgIsNotRunningIsOk)
 		if (kill(standalonePid, 0) == 0)
 		{
 			pgSetup->pidFile.pid = pid;
+			free(contents);
```

Comment:
I think we should unconditionally free the contents after a successful parse, and stop worrying in all the return cases thereafter.

Reply:
Sure, will do

Reply:
Let me know if this resolved with the latest push

Reply:
LGTM!